lib/active_merchant/billing/gateways/cecabank/cecabank_json.rb

```diff
@@ -174,7 +174,7 @@ def add_payment_method(post, payment_method, options)
         }
 
         if payment_method.is_a?(NetworkTokenizationCreditCard) && WALLET_PAYMENT_METHODS[payment_method.source.to_sym]
-          eci_to_format = payment_method.eci || (three_d_secure[:eci] if three_d_response)
+          eci_to_format = payment_method.eci || (three_d_secure[:eci] if three_d_secure)
```

Comment:
```suggestion
          eci_to_format = payment_method.eci || three_d_secure[:eci]
```

The if statement for `three_d_secure` is not needed because, in line 169, if the `three_d_secure` object is not present, the fetch retrieves an empty hash.